### ISSUE TITLE
Orca should announce the name of the file being edited when we use ctrl+PageUp or ctrl+pageDown to change editors.

### ISSUE THREAD

**Issue by jvesouza** [microsoft/vscode#141664]
Issue Type: <b>Bug</b>

CC: @isidorn

When I use ctrl+pageUp or ctrl+PageDown to change editors, orca does not announce the name of the file to be edited. While this isn't a big deal, it would be nice to hear the filename present in the editor.

I just can't say if this only happens with the orca or if it happens with other screen readers as well.

VS Code version: Code - Insiders 1.64.0-insider (3dda6ea7acb66ea8693032fe9631ccac629da770, 2022-01-27T05:15:20.649Z)
OS version: Linux x64 5.16.2-arch1-1
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-5200U CPU @ 2.20GHz (4 x 2494)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 1|
|Memory (System)|15.54GB (2.18GB free)|
|Process Argv|-n a.js b.js c.js --crash-reporter-id bcecf5e0-97d1-4830-b28c-0943c4adab1c|
|Screen Reader|yes|
|VM|0%|
|DESKTOP_SESSION|gnome-xorg|
|XDG_CURRENT_DESKTOP|GNOME|
|XDG_SESSION_DESKTOP|gnome-xorg|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (37)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-eslint|dba|2.2.2
EditorConfig|Edi|0.16.4
prettier-vscode|esb|9.1.0
shell-format|fox|7.2.2
rest-client|hum|0.24.6
vscode-docker|ms-|1.19.0
csharp|ms-|1.24.0
vscode-aks-tools|ms-|0.0.11
vscode-kubernetes-tools|ms-|1.3.6
python|ms-|2022.0.1719302831-dev
vscode-pylance|ms-|2022.1.4-pre.1
remote-containers|ms-|0.215.0
remote-ssh|ms-|0.70.0
remote-ssh-edit|ms-|0.70.0
azure-account|ms-|0.9.11
powershell|ms-|2021.12.0
vsliveshare|ms-|1.0.5273
vsliveshare-audio|ms-|0.1.91
oracledevtools|Ora|21.3.0
vscode-boot-dev-pack|Piv|0.1.0
vscode-concourse|Piv|1.29.0
vscode-manifest-yaml|Piv|1.29.0
vscode-spring-boot|Piv|1.29.0
fabric8-analytics|red|0.3.5
java|red|1.3.0
vscode-commons|red|0.0.6
vscode-xml|red|0.18.2
vscode-yaml|red|1.3.0
sonarlint-vscode|Son|3.2.0
vscodeintellicode|Vis|1.2.16
vscode-java-debug|vsc|0.38.0
vscode-java-dependency|vsc|0.19.0
vscode-java-pack|vsc|0.21.0
vscode-java-test|vsc|0.34.0
vscode-maven|vsc|0.35.0
vscode-spring-boot-dashboard|vsc|0.2.0
vscode-spring-initializr|vsc|0.8.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30422396
pythontb:30258533
pythonvspyt551cf:30291413
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
vscod805cf:30301675
pythonvspyt200:30323110
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30404738
wslgetstartedc:30421358
vscop804:30404766
vscop453:30404998
vsclayoutctrc:30405799
azactmsal:30409287
pythonvsnew555:30426298
vsbas813:30426126
vscscmwlcmc:30428972

```

</details>

**Comment by jvesouza** [comment]
I forgot to add that this also happens in VSCode version 1.63.

**Comment by isidorn** [comment]
@jvesouza thanks for filling this. I can reproduce this with Orca. NVDA and VoiceOver work fine.
To clarify: `ctrl+pageUp` and `ctrl+PageDown` change the editor to be the next or previous editor.

I believe the issue is coming from some previous heuristic we agreed that Orca should do so that the editor title is not read too often. @joanmarie might know better.
@joanmarie could Orca maybe check if the aria-label of the editor changed and then read it, if it is the same not read it. I understand this is easier said than done and I probably did not generalise the problem enough.

**Comment by joanmarie** [comment]
I'll take a look.

**Comment by joanmarie** [comment]
I just tried with version 1.63.2 (I now use VSCode stable for my Chromium development work). I can confirm Orca says nothing, but I am also not seeing *any* accessibility events (e.g. no name change, no selection change, no focus change, no ....). I'll try in insiders to make sure, but this may be an issue we need fixed in Chromium.

**Comment by isidorn** [comment]
@joanmarie awesome that we finally won you over ha!
We have a VS Code exploration build that uses Electron 16 that uses Chromium 96. We should in February also migrate Insiders to that version. In the meantime you can checkout the exploration build (maybe it already fixed the issue)
https://az764295.vo.msecnd.net/exploration/db4621a582f849d50f45d4826cb382a36720d002/code-exploration-x64-1643204890.tar.gz

**Comment by joanmarie** [comment]
Sorry for being spammy with contents, 1.64.0-insider Commit: 4af3e141de + Orca master: Orca announces each page switch for me.

**Comment by isidorn** [comment]
Yeah I was not using Orca master.
@jvesouza can you verify if this is indeed fixed with latest Orca master and then we can just close this.

**Comment by joanmarie** [comment]
Ugh. More spam from me (multitasking during a meeting. I suck): Orca was presenting the selection changes from the file explorer. When I hid that, I see the bug in insiders. And the bug is no events, so there's nothing Orca can do.

**Comment by isidorn** [comment]
@joanmarie no problem. Ok makes sense. Can you just double check with the exploration build and if it happens there also I guess we should file a chromium issue.

**Comment by joanmarie** [comment]
Exploration build segfaults upon launch with a not very useful stacktrace. Will need not-in-a-meeting-time for this I think.

**Comment by jvesouza** [comment]
@isidorn I tested using the master version of orca, but as already reported by @joanmarie  the problem persists.
If you can send me a link to an exploration build that runs on arch linux, I can test it too.
If there is a branch from which the exploration build is generated, I can try to generate a test version from that branch.

**Comment by isidorn** [comment]
@jvesouza I can send you a link to the exploration build. Just what flavour of VS Code do you use on linux? How many bits, what distribution?

**Comment by jvesouza** [comment]
$ uname -a
Linux jveii 5.16.2-arch1-1 #1 SMP PREEMPT Thu, 20 Jan 2022 16:18:29 +0000 x86_64
 GNU/Linux

64-bits
I always install from the .tar that is downloaded from the vscode insiders downloads page.

**Comment by joanmarie** [comment]
I just built from ToT and ran VSCode web in a recently-built-locally Chromium. I see the issue there too.

Would I get different results from the exploration build?

**Comment by isidorn** [comment]
@joanmarie no. If you run VS Code web in latest locally built chromium that means the bug is still present. And this is the best strategy to test.

**Comment by joanmarie** [comment]
Ok, I'll try to make some time for this next week.

**Comment by jvesouza** [comment]
When running this program nothing was shown. When consulting the logs I 
found dump records for code-exploration.

**Comment by joanmarie** [comment]
@jvesouza: I just committed a change to Orca master which hopefully addresses this. The problem is that the accessible object associated with editor you were in was destroyed and Orca was quietly recovering from that by finding the newly-focused object. The change I made to Orca is to actually announce the object in this case. There's a chance that can lead to chattiness in other web apps when the currently focused object is destroyed, so please test thoroughly. Thanks!

**Comment by isidorn** [comment]
@joanmarie thanks a lot for the quick fix. Let's see how it goes and what @jvesouza says.

**Comment by jvesouza** [comment]
@joanmarie  It is fixed, great job!

**Comment by isidorn** [comment]
Great. Closing this. Thanks @joanmarie 👏
